### PR TITLE
Add codecov

### PR DESCRIPTION
**Pull request by sidharthv96**
## :bookmark_tabs: Summary

Brief description about the content of your PR.

Resolves #<your issue id here>

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [ ] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added unit/e2e tests (if appropriate)
- [ ] :notebook: have added documentation (if appropriate)
- [ ] :bookmark: targeted `develop` branch

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -31,16 +31,23 @@ jobs:
 
       - name: Run Unit Tests
         run: |
-          pnpm run ci --coverage
+          pnpm test:coverage
```
Comment (Yokozuna59) [inline .github/workflows/test.yml:34]: https://github.com/mermaid-js/mermaid/pull/4500#discussion_r1233031021

Comment (sidharthv96) [reply .github/workflows/test.yml:34]: I think it'll automatically detect CI and run it properly.

**Comment by Yokozuna59** [comment]
@sidharthv96 should we relay on the old code coverage badge? I think it might be better to add codecov coverage badge on root README.md

**Comment by sidharthv96** [comment]
Yes, the old one can be removed. 
Can you raise the PR

**Comment by Yokozuna59** [comment]
Sure! but you have to go to settings and [enable it](https://app.codecov.io/gh/mermaid-js/mermaid/settings/badge) so it can be displayed. It's publicly exposed yet:

codecov

**Pull request merged**